Source/WebCore/dom/Element.cpp

```diff
@@ -2764,21 +2766,22 @@ void Element::setIsDefinedCustomElement(JSCustomElementInterface& elementInterfa
 
 void Element::setIsFailedCustomElement()
 {
-    setIsFailedCustomElementWithoutClearingReactionQueue();
-    clearReactionQueueFromFailedCustomElement();
+    ASSERT(isUnknownElement());
+    setIsFailedOrPrecustomizedCustomElementWithoutClearingReactionQueue();
+    clearReactionQueueFromFailedOrPrecustomizedCustomElement();
+    ASSERT(isFailedCustomElement());
 }
 
-void Element::setIsFailedCustomElementWithoutClearingReactionQueue()
+void Element::setIsFailedOrPrecustomizedCustomElementWithoutClearingReactionQueue()
 {
-    ASSERT(isUndefinedCustomElement());
     ASSERT(customElementState() == CustomElementState::Undefined);
-    setCustomElementState(CustomElementState::Failed);
+    setCustomElementState(CustomElementState::FailedOrPrecustomized);
     InspectorInstrumentation::didChangeCustomElementState(*this);
 }
 
-void Element::clearReactionQueueFromFailedCustomElement()
+void Element::clearReactionQueueFromFailedOrPrecustomizedCustomElement()
```

Comment:
I don't think we should rename this function.
Whenever we invoke this function, we know "this" is a failed custom element.